core-bundle/src/Resources/config/services.yml

```diff
@@ -379,6 +379,21 @@ services:
         arguments:
             - '@contao.menu.matcher'
 
+    contao.menu.frontend_builder:
+        class: Contao\CoreBundle\Menu\FrontendMenuBuilder
+        arguments:
+            - '@Knp\Menu\FactoryInterface'
+            - '@request_stack'
+            - '@event_dispatcher'
+            - '@database_connection'
+            - '@contao.routing.page_registry'
+            - '@=service("contao.framework").getAdapter("Contao\\PageModel")'
```

Comment:
We never did this so far but we sure can :) Should we? /cc @contao/developers

Reply:
Definitely yes. 👍

Reply:
Not sure this is bad for our tests?

Reply:
@richardhj Does this work reliably without calling `$this->framework->initialize()` in advance? Because now we are initializing the framework every time we call `$this->framework->getAdapter()`.

Reply:
Works like a charm.

The only issue I recognized is that [`$this->mockAdapter(...$methods)`](https://github.com/contao/test-case/blob/ceea79483c6040ab5d90c2b9cefd6a2ec5edf696/src/ContaoTestCase.php#L132) creates an Adapter instance without actual class. [`$this->class` is null](https://github.com/contao/contao/blob/6a815a472a46c44c4dbf3885f44c229a26fb8510/core-bundle/src/Framework/Adapter.php#L46) so that I decided to mock every method I use from that Adapter.

Another fix would be to update [`$this->mockAdapter(...$methods)`](https://github.com/contao/test-case/blob/ceea79483c6040ab5d90c2b9cefd6a2ec5edf696/src/ContaoTestCase.php#L132) to allow setting a class.